discord/gateway.py

```diff
@@ -649,6 +651,22 @@ def received_message(self, msg):
             yield from self.identify()
         elif op == self.SESSION_DESCRIPTION:
             yield from self.load_secret_key(data)
+        elif op == self.SPEAKING:
+            user_id = int(data['user_id'])
+            ssrc = int(data['ssrc'])
+
+            if 'speaking' in data:
+                user = self._connection._state.get_user(user_id)
+
+                self._dispatch('voice_speaking_state', self._connection, user, data['speaking'])
+
+                # Used for continuous streams
+                for f in self._connection._speaking_listeners:
+                    f(user, data['speaking'])
+
+                if self._connection.channel.id not in self._connection._ssrc_lookup:
+                    self._connection._ssrc_lookup[self._connection.channel.id] = {}
```

Comment:
you'd really want to key this by guild id - not channel id. ssrc's are shared across channels in a guild.